### PR TITLE
Blog - Caching Dependencies on GitHub Actions

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
Preview URL: https://6414c12765f55019fd2045df--ephemeral-narwhal-3ac3b2.netlify.app

File: blog/_posts/2023-04-28-caching-dependencies-github-actions.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+Caching is a good fit for saving and reusing files that don't change too often such as third-party dependencies. In comparison, artifacts are more suitable for when you want to save and potentially download files that were generated as the output of a job, such as log files, binaries, and application builds. This article will focus only on caching, but it's important to know the difference.
+
+Another thing you need to know prior to starting this tutorial is how access restrictions work for caches. Generally speaking, a workflow run will only be able to access a cache that was created in the current branch, the base branch of a pull request, or the default branch of the repository (or the repository it was forked from). In practice, this means that if you have a branch called `feature-b` that is based on another branch, `feature-a`, which in turn was based on the default branch `main`, a workflow run in `feature-b` will be able to access caches created in `feature-b`, `feature-a`, and `main`. Whereas a workflow run in `feature-a` will only be able to access caches created in `feature-a` and `main`. You can see a visual representation of this below:
+
```
Comment (balapriyac) [inline blog/_posts/2023-04-28-caching-dependencies-github-actions.md:46]:
```suggestion

Another thing you need to know prior to starting this tutorial is how access restrictions work for caches. Generally speaking, a workflow run will only be able to access a cache that was created in the current branch, the base branch of a pull request, or the default branch of the repository (or the repository it was forked from). 

In practice, this means that if you have a branch called `feature-b` that is based on another branch, `feature-a`, which in turn was based on the default branch `main`, a workflow run in `feature-b` will be able to access caches created in `feature-b`, `feature-a`, and `main`. Whereas a workflow run in `feature-a` will only be able to access caches created in `feature-a` and `main`. You can see a visual representation of this below:

```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer!

**Comment by msali123** [comment]
Thanks @balapriyac 

Changes done: https://641956774638565bdcbc7711--ephemeral-narwhal-3ac3b2.netlify.app/blog/caching-dependencies-github-actions/

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Review by jalletto — APPROVED** [review]

**Pull request merged**